Shared/Samples/Play KML tour/PlayKMLTourView.swift

```diff
@@ -0,0 +1,143 @@
+    @State private var viewpoint: Viewpoint?
+    
+    /// The tour controller used to control the KML tour.
+    @State private var tourController = KMLTourController()
+    
+    /// The current status of the KML tour.
+    @State private var tourStatus: KMLTour.Status = .notInitialized
+    
+    /// A Boolean that indicates whether to show an error alert.
+    @State private var isShowingErrorAlert = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error? {
+        didSet { isShowingErrorAlert = error != nil }
+    }
+    
+    /// A Boolean value that indicates whether to disable the tour buttons.
+    private var tourDisabled: Bool {
+        tourStatus == .notInitialized || tourStatus == .initializing
+    }
+    
+    var body: some View {
+        // Create a scene view with a scene and a viewpoint.
+        SceneView(scene: scene, viewpoint: viewpoint)
+            .onViewpointChanged(kind: .centerAndScale) { viewpoint = $0 }
+            .task {
+                do {
+                    // Add KML layer to the scene.
+                    let dataset = KMLDataset(url: .esriTour)
+                    try await dataset.load()
+                    let layer = KMLLayer(dataset: dataset)
+                    scene.addOperationalLayer(layer)
+                    
+                    // Set the tour controller with first tour in the dataset.
+                    tourController.tour = dataset.tours.first
+                    
+                    // Listen for tour status updates.
+                    if let tour = tourController.tour {
+                        for await status in tour.$status {
+                            tourStatus = status
+                        }
+                    }
+                } catch {
+                    self.error = error
+                }
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    ZStack {
+                        HStack {
+                            Button {
+                                tourController.reset()
+                                viewpoint = scene.initialViewpoint
+                            } label: {
+                                Image(systemName: "gobackward")
+                            }
+                            .disabled(tourDisabled || tourStatus == .initialized)
+                            Spacer()
+                        }
+                        
+                        Button {
+                            tourStatus == .playing ? tourController.pause() : tourController.play()
+                        } label: {
+                            Image(systemName: tourStatus == .playing ? "pause.fill" : "play.fill")
+                        }
+                        .disabled(tourDisabled)
+                    }
+                }
+            }
+            .alert(isPresented: $isShowingErrorAlert, presentingError: error)
```

Comment:
```suggestion
            .alert(isPresented: $isShowingErrorAlert, presentingError: error)
            .overlay(alignment: .center) {
                if tourDisabled {
                    VStack {
                        Text("Initializing KML Tour…")
                        ProgressView()
                            .progressViewStyle(.circular)
                    }
                    .padding()
                    .background(.ultraThickMaterial)
                    .cornerRadius(10)
                    .shadow(radius: 50)
                }
            }
```

I suggest adding a progress view to indicate that the tour is not immediately accessible. Feel free to change the text. 

Here is another option that blurs the background until the tour is ready:

```swift
.overlay(alignment: .center) {
                if tourDisabled {
                    ZStack {
                        Color.clear.background(.ultraThinMaterial)
                        VStack {
                            Text("Initializing KML Tour…")
                            ProgressView()
                                .progressViewStyle(.circular)
                        }
                        .padding()
                        .background(.ultraThickMaterial)
                        .cornerRadius(10)
                        .shadow(radius: 50)
                    }
                }
            }
```

Reply:
I added some text in addition to the your progress view so that the user can always see the current tour status. Let me know what you think, and I'll update the screenshot.

Reply:
Looks good. Please update the screenshot.